src/Symfony/Component/DependencyInjection/Compiler/AnalyzeServiceReferencesPass.php

```diff
@@ -74,6 +74,9 @@ public function process(ContainerBuilder $container)
             if ($definition->getFactoryService()) {
                 $this->processArguments(array(new Reference($definition->getFactoryService())));
             }
+            if (is_array($definition->getFactory())) {
```

Comment:
hmm, sorry, this is about resolving references. The name ``processArguments`` is confusing